examples/Attic & Button/config/routes.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+#
+#    Copyright 2018, Optimizely and contributors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+Rails.application.routes.draw do
+  resources :demo, only: %i[new create]
+
+  # Binding user_id auth requests
+  post 'demo/:user_id/buy', to: 'demo#buy', as: :buy
+  get 'demo/:user_id/cart', to: 'demo#cart', as: :cart
+  get 'demo/:user_id/checkout_cart', to: 'demo#checkout_cart', as: :checkout_cart
+  post 'demo/:user_id/checkout_payment', to: 'demo#checkout_payment', as: :checkout_payment
+  delete 'demo/:user_id/delete_messages', to: 'demo#delete_messages', as: :delete_messages
+  delete 'demo/:user_id/delete_cart', to: 'demo#delete_cart', as: :delete_cart
+  get 'demo/:user_id/messages', to: 'demo#log_messages', as: :messages
+  get '/demo/:user_id/payment' => 'demo#payment', as: :payment
+  get '/demo/:user_id/shop' => 'demo#shop', as: :shop
+  put '/demo/:user_id/update_cart' => 'demo#update_cart', as: :update_cart
+  get '/show_receipt/:user_id/show_receipt' => 'demo#show_receipt', as: :show_receipt
+
+  # Requests before user logged in
+  post 'demo/config', to: 'demo#create'
+  get 'demo/login', to: 'demo#new'
+  delete 'demo/logout', to: 'demo#logout', as: :logout
+  get '/demo/shop' => 'demo#guest_shop', as: :guest_shop
+
```

Comment:
Please order routes as per the flow of demo app. And write a short comment for each route of what it does, or what page it leads to

Reply:
Done